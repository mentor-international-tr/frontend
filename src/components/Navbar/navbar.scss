@import "src/lib/styles/variables.scss";

.mi-navbar__list {
    display: flex;
    justify-content: space-between;
    max-width: 600px;
}

.mi-navbar__list-item {
    list-style: none;
}