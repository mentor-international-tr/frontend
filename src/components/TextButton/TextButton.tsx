import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";

const styles = (theme: any) => ({
  button: {
    margin: theme.spacing.unit,
  },
  input: {
    display: "none",
  },
});

export interface TextButtonProps {
  classes: any;
  children: React.ReactChild;
  btnProps?: object;
}

const TextButton: React.SFC<TextButtonProps> = ({
  classes,
  children,
  btnProps,
}) => {
  return (
    <Button className={classes.button} {...btnProps}>
      {children}
    </Button>
  );
};
TextButton.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(TextButton);
