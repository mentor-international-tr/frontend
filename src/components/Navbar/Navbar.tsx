import "./navbar.scss";
import React from "react";
import uuidv4 from "uuid/v4";
export interface NavbarProps {
  children: any;
}

const Navbar: React.SFC<NavbarProps> = ({ children }) => {
  return (
    <nav className="mi-navbar">
      <ul className="mi-navbar__list">
        {children.map((child: any) => {
          return (
            <li
              key={uuidv4()}
              className="mi-navbar__list-item"
            >
              {React.cloneElement(child)}
            </li>
          );
        })}
      </ul>
    </nav>
  );
};

export default Navbar;
