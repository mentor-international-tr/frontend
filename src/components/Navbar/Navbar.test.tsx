import React from "react";
import { render, cleanup } from "react-testing-library";
import Navbar from "./Navbar";
import "jest-dom/extend-expect";
import TextButton from "components/TextButton/TextButton";

afterEach(cleanup);

describe("testing <Navbar /> component", () => {
  it("creates snapshot", () => {
    const { asFragment, rerender } = render(
      <Navbar>
          <TextButton>Message</TextButton>
          <TextButton>Templates</TextButton>
          <TextButton>People</TextButton>
      </Navbar>,
    );
    expect(asFragment()).toMatchSnapshot();
    })
})
