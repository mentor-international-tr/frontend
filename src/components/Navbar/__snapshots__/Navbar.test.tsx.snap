// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Navbar /> component creates snapshot 1`] = `
<DocumentFragment>
  <nav
    class="mi-navbar"
  >
    <ul
      class="mi-navbar__list"
    >
      <li
        class="mi-navbar__list-item"
      >
        <button
          class="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-text-5 MuiButton-flat-8 TextButton-button-1"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-4"
          >
            Message
          </span>
          <span
            class="MuiTouchRipple-root-32"
          />
        </button>
      </li>
      <li
        class="mi-navbar__list-item"
      >
        <button
          class="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-text-5 MuiButton-flat-8 TextButton-button-1"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-4"
          >
            Templates
          </span>
          <span
            class="MuiTouchRipple-root-32"
          />
        </button>
      </li>
      <li
        class="mi-navbar__list-item"
      >
        <button
          class="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-text-5 MuiButton-flat-8 TextButton-button-1"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-4"
          >
            People
          </span>
          <span
            class="MuiTouchRipple-root-32"
          />
        </button>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
